### PR TITLE
Fix the size of predicted trajectories during evaluation

### PR DESCRIPTION
**Pull request by qiyan98**
The `len(dataset)` input to the `merge_results_dist` function incorrectly restricts the size of the merged `pred_dicts`. It should be equal to the total number of trajectories (center objects) in the validation set, as each element in `pred_dicts` corresponds to a single trajectory. However, using `len(dataset)` actually represents the **number of scenarios**, which is significantly smaller than the **number of center objects**.

In practice, we have observed that the original code consistently saves the results of 40,097 trajectories in the `result.pkl` file, where 40,097 is the number of scenarios in the validation set. Nevertheless, we should ideally have 192,172 trajectories accounted for for the validation set. This discrepancy results in the effective validation of smaller prediction outcomes.

### CODE REVIEW

**Comment by sshaoshuai** [comment]
Thanks very much for reporting the bug! It is indeed a bug caused during the code refactor process of the dataloader. I have fixed it in this PR: https://github.com/sshaoshuai/MTR/pull/63.  

Note that we can not simply change the `size` argument of `merge_results_dist` to MAX_INT, since this argument is used to clip duplicate items when len(dataset) can not be evenly divided by the number of GPUs, and we need to set it to len(dataset).
